### PR TITLE
8-Implement-Delete-Command

### LINKED ISSUE #8: As a recruiter I can delete outdated job listings.
So that my listings are up to date.

### PR DESCRIPTION
**Pull request by nt-nic**
## Which ticket does this address

- closes #8 

## What problems does this solve

- No command to delete job listings

## Solution/Changes

- Add DeleteCommand class which deletes job listings from the listing book
- Add DeleteCommandParser class which parses user input to extract arguments for DeleteCommand
- Add DeleteCommandTest class to run tests on DeleteCommand class
- Add DeleteCommandParserTest class to run tests in DeleteCommandParser class

## Checklist

- [ ] All tests pass locally and on CI

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -7,7 +7,7 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_LISTING_DISPLAYED_INDEX = "The listing index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
```
Comment (yyj-02) [inline src/main/java/seedu/address/commons/core/Messages.java:11]: Can we change this to MESSAGE_LISTINGS...? If it is not within the same scope it is okay

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Test
+    public void parse_validArgs_returnsDeleteCommand() {
+        assertParseSuccess(parser, "1", new DeleteCommand(INDEX_FIRST_LISTING));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "a", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                DeleteCommand.MESSAGE_USAGE));
+    }
```
Comment (yyj-02) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:31]: add one more test case for out of index, i.e. -1 or something

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredListingList().size() + 1);
+        DeleteCommand deleteCommand = new DeleteCommand(outOfBoundIndex);
+
+        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_LISTING_DISPLAYED_INDEX);
+    }
+
```
Comment (yyj-02) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:54]: one more test case for passing invalid character -> a

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public class DeleteCommandParser implements Parser<DeleteCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteCommand
+     * and returns a DeleteCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteCommand parse(String args) throws ParseException {
+        try {
+            Index index = ParserUtil.parseIndex(args);
```
Comment (yyj-02) [inline src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:21]: do we need to check if args is empty? I am not too sure, but can you check previous implementation

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (yyj-02) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:32]: add a second test case passing empty string

**Review by yyj-02 — CHANGES_REQUESTED** [review]
Just add some test cases

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java (hunk 4 above)
Comment (nt-nic) [reply src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:21]: Empty args case is handled by ParserUtil.parseIndex(), will throw a ParseException

Previous implementation also did not account for empty args here

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java (hunk 5 above)
Comment (nt-nic) [reply src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:36]: test case added

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java (hunk 3 above)
Comment (nt-nic) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:54]: Handled by DeleteCommandParser, in this case the DeleteCommand will not be instantiated

**Review by yyj-02 — APPROVED** [review]
lgtm

**Review by panpannnnn — APPROVED** [review]
lgtm
